src/main/groovy/com/morpheusdata/nutanix/prismelement/plugin/sync/ContainersSync.groovy

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2024 Morpheus Data, LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.morpheusdata.nutanix.prismelement.plugin.sync
+
+import com.morpheusdata.core.MorpheusContext
+import com.morpheusdata.core.data.DataQuery
+import com.morpheusdata.core.util.HttpApiClient
+import com.morpheusdata.core.util.SyncTask
+import com.morpheusdata.model.Cloud
+import com.morpheusdata.model.Datastore
+import com.morpheusdata.model.projection.DatastoreIdentity
+import com.morpheusdata.nutanix.prismelement.plugin.NutanixPrismElementPlugin
+import com.morpheusdata.nutanix.prismelement.plugin.utils.NutanixPrismElementApiService
+import groovy.util.logging.Slf4j
+
+/**
+ * A class to sync Nutanix Storage Containers to their Morpheus Datastore equivalent
+ */
+@Slf4j
+class ContainersSync {
+	private final MorpheusContext context
+	private final HttpApiClient client
+	private final Cloud cloud
+
+	ContainersSync(MorpheusContext context, Cloud cloud, HttpApiClient client) {
+		this.context = context
+		this.cloud = cloud
+		this.client = client
+	}
+
+	def execute() {
+		log.info("Executing container sync for cloud ${cloud.name}")
+		try {
+			def authConfig = NutanixPrismElementPlugin.getAuthConfig(context, cloud)
+			def listResults = NutanixPrismElementApiService.listContainers(client, authConfig)
+			if (listResults.success == true) {
+				def objList = listResults?.containers?.findAll {
+					it.name?.toLowerCase() != 'nutanixmanagementshare'
+				}
+
+				def existingItems = context.async.cloud.datastore.listIdentityProjections(
+					new DataQuery()
+						.withFilter('refType', 'ComputeZone')
+						.withFilter('refId', cloud.id)
+						.withFilter('type', 'generic')
+				)
+
+				SyncTask<DatastoreIdentity, Map, Datastore> syncTask = new SyncTask<>(existingItems, objList)
+				syncTask.addMatchFunction { DatastoreIdentity datastoreIdentity, Map cloudItem ->
+					datastoreIdentity?.externalId == cloudItem?.uuid
+				}.onAdd { List<Map> cloudItems ->
+					cloudItems?.each { cloudItem ->
```

Comment:
Collect a list of adds and use `context.services.cloud.datastore.bulkCreate(adds)` to save on API call overhead.